portal/tasks.py

```diff
@@ -182,41 +212,81 @@ def update_patient_loop(update_cache=True, queue_messages=True):
     valid_patients = User.query.join(UserRoles).filter(and_(
         User.id == UserRoles.user_id,
         User.deleted_id.is_(None),
-        UserRoles.role_id == patient_role_id))
+        UserRoles.role_id == patient_role_id)).with_entities(User.id)
+
+    patients = valid_patients.all()
+    j = 0
+    batchsize = 16
+    while True:
```

Comment:
this is [my favorite way to chunk lists](https://stackoverflow.com/a/435712) I've found, 

ie:
```
for chunk in zip(*[iter(patients)]*batchsize):
...
```
[this might be better though ](https://stackoverflow.com/a/8312908)(doesn't skip the "remainder" if `patients` is not divisible by chunk size)

Reply:
not sure that makes the code any easier to read, we'd like to process the whole list, and then i need to make the consumer None safe...

I'll change it if you want it, tho?

Reply:
I'm just paranoid about infinite loops, no need to change